### PR TITLE
EKS Observability Accelerator - Memcached

### PR DESCRIPTION
**Pull request by saaish**
### More

- [ ] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: examples/observability/eks-memcached/README.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+aws eks --region $AWS_REGION update-kubeconfig --name aws001-preprod-observability-eks
+```
+
+`terraform apply` will provision all the aforementioned resources.
+
+---
+
+**NOTE**
+
+This example deploy all the necessary components to start monitoring your Java-based
```
Comment (askulkarni2) [inline examples/observability/eks-memcached/README.md:89]: Memcached

File: modules/kubernetes-addons/adot-collector-memcached/locals.tf [hunk 2]
```diff
@@ -0,0 +1,60 @@
+locals {
+  name                           = "adot-collector"
+  adot_collector_service_account = "adot-collector"
```
Comment (askulkarni2) [inline modules/kubernetes-addons/adot-collector-memcached/locals.tf:3]: `adot-collector-memcached`

File: modules/kubernetes-addons/adot-collector-memcached/otel-config/templates/clusterrole.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    resources:
+      - ingresses
+    verbs:
+      - get
+      - list
+      - watch
+  - nonResourceURLs:
+      - /metrics
+    verbs:
+      - get
```
Comment (askulkarni2) [inline modules/kubernetes-addons/adot-collector-memcached/otel-config/templates/clusterrole.yaml:29]: newline

File: modules/kubernetes-addons/adot-collector-memcached/otel-config/templates/clusterrolebinding.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: otel-prometheus-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: otel-prometheus-role
+subjects:
+  - kind: ServiceAccount
+    name: adot-collector
+    namespace: adot-collector
```
Comment (askulkarni2) [inline modules/kubernetes-addons/adot-collector-memcached/otel-config/templates/clusterrolebinding.yaml:12]: `adot-collector-memcached`

File: modules/kubernetes-addons/main.tf [hunk 5]
```diff
@@ -306,3 +306,39 @@ module "adot_collector_java" {
...
+
+module "adot_collector_haproxy" {
+  count                                = var.enable_adot_collector_haproxy ? 1 : 0
+  source                               = "./adot-collector-haproxy"
+  helm_config                          = var.adot_collector_haproxy_helm_config
+  amazon_prometheus_workspace_endpoint = var.amazon_prometheus_workspace_endpoint
+  amazon_prometheus_workspace_region   = var.amazon_prometheus_workspace_region
+  addon_context                        = local.addon_context
+}
+
```
Comment (askulkarni2) [inline modules/kubernetes-addons/main.tf:336]: Remove this

File: modules/kubernetes-addons/variables.tf [hunk 6]
```diff
@@ -711,3 +711,38 @@ variable "adot_collector_java_helm_config" {
...
+  type        = bool
+  default     = false
+  description = "Enable metrics for HAProxy workloads"
+}
+
+variable "adot_collector_haproxy_helm_config" {
+  type        = any
+  default     = {}
+  description = "ADOT Collector HAProxy Helm Chart config"
+}
```
Comment (askulkarni2) [inline modules/kubernetes-addons/variables.tf:737]: Remove

**Review by askulkarni2 — COMMENTED** [review]
Similar comments to other PRs.

File: examples/observability/eks-memcached/README.md (hunk 1 above)
Comment (saaish) [reply examples/observability/eks-memcached/README.md]: changed

File: modules/kubernetes-addons/adot-collector-memcached/locals.tf (hunk 2 above)
Comment (saaish) [reply modules/kubernetes-addons/adot-collector-memcached/locals.tf]: added

File: modules/kubernetes-addons/adot-collector-memcached/otel-config/templates/clusterrole.yaml (hunk 3 above)
Comment (saaish) [reply modules/kubernetes-addons/adot-collector-memcached/otel-config/templates/clusterrole.yaml]: done

File: modules/kubernetes-addons/main.tf (hunk 5 above)
Comment (saaish) [reply modules/kubernetes-addons/main.tf]: done

File: modules/kubernetes-addons/variables.tf (hunk 6 above)
Comment (saaish) [reply modules/kubernetes-addons/variables.tf]: modified

**Review by askulkarni2 — APPROVED** [review]

**Review by bryantbiggs — APPROVED** [review]
Thank you @saaish !

**Pull request merged**